core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/PEHelper64.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2023 QNX Software Systems and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     QNX Software Systems - Initial API and implementation of ElfHelper
+ *     John Dallaway - Initial PEHelper64 implementation (#652)
+ *******************************************************************************/
+package org.eclipse.cdt.utils.coff;
+
+import java.io.IOException;
+
+import org.eclipse.cdt.utils.coff.Coff64.FileHeader;
+import org.eclipse.cdt.utils.coff.Coff64.SectionHeader;
+
+/**
+ * <code>PEHelper64</code> is a wrapper class for the <code>PE64</code> class
+ * to provide higher level API for sorting/searching the COFF data.
+ *
+ * @see PE64
+ * @since 8.4
+ */
+public class PEHelper64 {
+
+	public static class Sizes {
```

Comment:
Please consider [record](https://docs.oracle.com/en/java/javase/14/language/records.html#GUID-6699E26F-4A9B-4393-A08B-1E47D4B2D263) construct, it could fit here

Reply:
Yes, much cleaner.